server/methods/createDirectMessage.js

```diff
@@ -8,74 +8,91 @@ import { RateLimiter } from '../../app/lib';
+		let to = Users.findOneByUsernameIgnoringCase(username);
+
+		// If the username does have an `@`, but does not exist locally, we create it first
+		if (!to && username.indexOf('@') !== -1) {
+			to = addUser(username);
 		}
 
-		if (settings.get('Message_AllowDirectMessagesToYourself') === false && usernames.length === 1 && me.username === usernames[0]) {
+		if (!to) {
 			throw new Meteor.Error('error-invalid-user', 'Invalid user', {
 				method: 'createDirectMessage',
 			});
 		}
+		return to;
+	});
 
-		const users = usernames.filter((username) => username !== me.username).map((username) => {
-			let to = Users.findOneByUsernameIgnoringCase(username);
+	const roomUsers = excludeSelf ? users : [me, ...users];
 
-			// If the username does have an `@`, but does not exist locally, we create it first
-			if (!to && username.indexOf('@') !== -1) {
-				to = addUser(username);
-			}
+	if (roomUsers.length === 1) {
+		throw new Meteor.Error('error-invalid-user', 'Invalid user', {
+			method: 'createDirectMessage',
+		});
+	}
+
+	if (!hasPermission(Meteor.userId(), 'create-d')) {
+		// If the user can't create DMs but can access already existing ones
+		if (hasPermission(Meteor.userId(), 'view-d-room')) {
+			// Check if the direct room already exists, then return it
 
-			if (!to) {
-				throw new Meteor.Error('error-invalid-user', 'Invalid user', {
-					method: 'createDirectMessage',
-				});
+			const uids = roomUsers.map(({ _id }) => _id).sort();
+			const room = Rooms.findOneDirectRoomContainingAllUserIDs(uids, { fields: { _id: 1 } });
+			if (room) {
+				return {
+					t: 'd',
+					rid: room._id,
+					...room,
+				};
 			}
-			return to;
+		}
+
+		throw new Meteor.Error('error-not-allowed', 'Not allowed', {
+			method: 'createDirectMessage',
 		});
+	}
 
-		if (!hasPermission(Meteor.userId(), 'create-d')) {
-			// If the user can't create DMs but can access already existing ones
-			if (hasPermission(Meteor.userId(), 'view-d-room')) {
-				// Check if the direct room already exists, then return it
-
-				const uids = [me, ...users].map(({ _id }) => _id).sort();
-				const room = Rooms.findOneDirectRoomContainingAllUserIDs(uids, { fields: { _id: 1 } });
-				if (room) {
-					return {
-						t: 'd',
-						rid: room._id,
-						...room,
-					};
-				}
-			}
+	const { _id: rid, inserted, ...room } = createRoom('d', null, null, roomUsers, null, { }, { creator: me._id });
 
-			throw new Meteor.Error('error-not-allowed', 'Not allowed', {
-				method: 'createDirectMessage',
-			});
-		}
+	return {
+		t: 'd',
+		rid,
+		...room,
+	};
+}
 
-		const { _id: rid, inserted, ...room } = createRoom('d', null, null, [me, ...users], null, { }, { creator: me._id });
+Meteor.methods({
+	createDirectMessage(...usernames) {
+		return createDirectMessage(usernames, false);
+	},
 
-		return {
-			t: 'd',
-			rid,
-			...room,
-		};
+	createDirectMessageForOthers(...usernames) {
```

Comment:
I'd ask to not create a new Meteor method just for internal usage. Instead just export the `createDirectMessage` function and call it when needed.